core-processor/src/processor.rs

```diff
@@ -42,33 +49,197 @@ enum SuccessfulDispatchResultKind {
     Success,
 }
 
-/// Process program & dispatch for it and return journal for updates.
-pub fn process<A: ProcessorExt + EnvExt + IntoExtInfo + 'static, E: Environment<A>>(
+/// Checked parameters for message execution across processing runs.
+pub struct PreparedMessageExecutionContext {
```

Comment:
I'd also keep here and in other places something like 
```rust
struct Actor {
    id: ProgramId,
    balance: u128,
    executable_data: Option<(Program, MemoryPages)>
}
```

Reply:
To tell the truth I don't see any profit to introduce such structure currently. There is exactly mentioned entity but it is used to distingiush between executable and not executable program.